pkg/chunk/cache/background.go

```diff
@@ -33,7 +33,7 @@ type BackgroundConfig struct {
 // RegisterFlagsWithPrefix adds the flags required to config this to the given FlagSet
 func (cfg *BackgroundConfig) RegisterFlagsWithPrefix(prefix string, description string, f *flag.FlagSet) {
 	f.IntVar(&cfg.WriteBackGoroutines, prefix+"memcache.write-back-goroutines", 10, description+"How many goroutines to use to write back to memcache.")
-	f.IntVar(&cfg.WriteBackBuffer, prefix+"memcache.write-back-buffer", 10000, description+"How many chunks to buffer for background write back.")
+	f.IntVar(&cfg.WriteBackBuffer, prefix+"memcache.write-back-buffer", 10000, description+"How many key batches to buffer for background write-back.")
```

Comment:
Currently this means we can buffer 10000 * 100 = 1Mil chunks which is not something we want? Can we reduce it to `1000/100` = 10 batches at once? But this further means that if there are 10 writes each with a single key, then we'll only buffer 10 keys.

Not sure how to handle this simply.

Reply:
This is not a new problem; previously we could buffer 10000*(any number) chunks.
Suggest we deal with your point separately.